core/store/src/adapter/epoch_store.rs

```diff
@@ -0,0 +1,75 @@
+use borsh::BorshDeserialize;
+use near_chain_primitives::Error;
+use near_primitives::epoch_block_info::BlockInfo;
+use near_primitives::epoch_info::EpochInfo;
+use near_primitives::epoch_manager::AGGREGATOR_KEY;
+use near_primitives::epoch_sync::{EpochSyncProof, EpochSyncProofV1};
+use near_primitives::errors::EpochError;
+use near_primitives::hash::CryptoHash;
+use near_primitives::types::{BlockHeight, EpochId};
+
+use crate::{DBCol, Store};
+
+use super::StoreAdapter;
+
+#[derive(Clone)]
+pub struct EpochStoreAdapter {
+    store: Store,
+}
+
+impl StoreAdapter for EpochStoreAdapter {
+    fn store_ref(&self) -> &Store {
+        &self.store
+    }
+}
+
+impl EpochStoreAdapter {
+    pub fn new(store: Store) -> Self {
+        Self { store }
+    }
+
+    pub fn get_epoch_start(&self, epoch_id: &EpochId) -> Result<BlockHeight, EpochError> {
+        self.store
+            .get_ser::<BlockHeight>(DBCol::EpochStart, epoch_id.as_ref())?
+            .ok_or(EpochError::EpochOutOfBounds(*epoch_id))
+    }
+
+    /// Get BlockInfo for a block
+    /// Errors
+    ///   EpochError::IOErr if storage returned an error
+    ///   EpochError::MissingBlock if block is not in storage
+    pub fn get_block_info(&self, hash: &CryptoHash) -> Result<BlockInfo, EpochError> {
+        self.store
+            .get_ser::<BlockInfo>(DBCol::BlockInfo, hash.as_ref())?
+            .ok_or(EpochError::MissingBlock(*hash))
+    }
+
+    pub fn get_epoch_info(&self, epoch_id: &EpochId) -> Result<EpochInfo, EpochError> {
+        self.store
+            .get_ser::<EpochInfo>(DBCol::EpochInfo, epoch_id.as_ref())?
+            .ok_or(EpochError::EpochOutOfBounds(*epoch_id))
+    }
+
+    // Iterate over all the epoch infos in store
+    pub fn iter_epoch_info<'a>(&'a self) -> Box<dyn Iterator<Item = (EpochId, EpochInfo)> + 'a> {
```

Comment:
I think we can avoid using `Box` + `dyn` here and just return `impl Iterator`, but feel free to merge as it is for now, I will try that and submit a separate PR if it works

Reply:
Nice, you're right! I can do away with the Box and just use impl Iterator. I was originally going off what we had in `DBIterator<'a>`